### PR TITLE
Add a spec file for Pulp platform to Fedora 23.

### PR DESCRIPTION
**Pull request by bowlofeggs**

### CODE REVIEW

File: rpms/pulp/pulp.spec [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+BuildRequires: selinux-policy-devel
+BuildRequires: systemd
+
+
+%description
+Pulp provides replication, access, and accounting for software repositories.
+
+
+%prep
+%autosetup -n %{name}-%{name}-%{version}-1
```
Comment (jeremycline) [inline rpms/pulp/pulp.spec:36]: It's pulp all the way down

File: rpms/pulp/pulp.spec [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+
+# These files will be generated during installation
+touch %{buildroot}/%{_sysconfdir}/pki/%{name}/ca.crt
+touch %{buildroot}/%{_sysconfdir}/pki/%{name}/ca.key
+touch %{buildroot}/%{_sysconfdir}/pki/%{name}/rsa.key
+touch %{buildroot}/%{_sysconfdir}/pki/%{name}/rsa_pub.key
+
+# pulp-streamer installation
+install -d %{buildroot}/%{_sysconfdir}/default/
+install -d %{buildroot}/%{_sysconfdir}/httpd/conf.d/
```
Comment (jeremycline) [inline rpms/pulp/pulp.spec]: It looks like this got made earlier as part of the "multiple sub-packages use these dirs" section. Doesn't hurt anything to be here, but it should be able to go as well. Same goes for a few other dirs in this block.

File: rpms/pulp/pulp.spec [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+%{_bindir}/%{name}-consumer
+%dir %{_usr}/lib/%{name}/consumer
+%{python2_sitelib}/%{name}/client/consumer/
+%{python2_sitelib}/pulp_client_consumer*.egg-info
+%doc %{_mandir}/man1/pulp-consumer.1*
+
+%defattr(640,root,root,-)
+%ghost %{_sysconfdir}/pki/%{name}/consumer/rsa.key
+
+%post consumer-client
```
Comment (jeremycline) [inline rpms/pulp/pulp.spec:314]: The permissions are not getting set correctly here. umask should be used.

File: rpms/pulp/pulp.spec [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+%pre server
+# If we are upgrading
+if [ $1 -gt 1 ] ; then
+        /bin/systemctl stop pulp_workers > /dev/null 2>&1
+        /bin/systemctl stop pulp_celerybeat > /dev/null 2>&1
+        /bin/systemctl stop pulp_resource_manager > /dev/null 2>&1
+fi
+
+
+%post server
```
Comment (jeremycline) [inline rpms/pulp/pulp.spec:622]: umask should be used here, too

File: rpms/pulp/pulp.spec [hunk 5]
```diff
@@ -0,0 +1,853 @@
...
+%{?python_provide:%python_provide python2-client-lib}
+
+
+%description -n python2-pulp-client-lib
+A framework for loading Pulp client extensions.
+
+
+%files -n python2-pulp-client-lib
+%license LICENSE
+%dir %{python2_sitelib}/%{name}/client
```
Comment (jeremycline) [inline rpms/pulp/pulp.spec:724]: If the package owns a dir, does it not automatically own all the files within the dir?

**Comment by jeremycline** [comment]
Other than the issues noted, LGTM. The only two comments I feel strongly about are the umask ones, everything else is related to tidying so I'm fine with that happening later if you want.

File: rpms/pulp/pulp.spec (hunk 1 above)
Comment (bowlofeggs) [inline rpms/pulp/pulp.spec:36]: repos/repos/pulp/pulp/repos/pulp/repos

File: rpms/pulp/pulp.spec (hunk 2 above)
Comment (bowlofeggs) [inline rpms/pulp/pulp.spec]: Fixed, and I have no idea how you noticed that.

File: rpms/pulp/pulp.spec (hunk 3 above)
Comment (bowlofeggs) [inline rpms/pulp/pulp.spec:314]: Sigh, this hasn't been fixed in Fedora yet and I copied this spec from there. It's fixed here and there now.

File: rpms/pulp/pulp.spec (hunk 4 above)
Comment (bowlofeggs) [inline rpms/pulp/pulp.spec:622]: I will claim the same excuse here too. Fixed.

File: rpms/pulp/pulp.spec (hunk 5 above)
Comment (bowlofeggs) [inline rpms/pulp/pulp.spec:724]: No, I think that's why the %dir macro exists. A package can create and own a directory that is used by other packages. For example, a package could create a folder that plugins should install filed into.

**Pull request merged**